### PR TITLE
feat: mount procfs instead of using hostpid (#1688)

### LINKED ISSUE #1688: mount procfs instead of using hostpid
## Feature Request

KubeArmor needs access to host's procfs to stich container and real time process information we get from eBPF. 

**Is your feature request related to a problem? Please describe the use case.**

Sharing pid ns with host is more risky than just accessing procfs.

**Describe the solution you'd like**

- We can mount and set procfs to readonly maybe alleviating a lot of risks involded with mounting procfs
- Add configuration in KubeArmor to support custom path prefix for procfs

Related #1186

### PR DESCRIPTION
**Pull request by yp969803**
Signed-off-by: Yash Patel <yp969803@gmail.com>

**Purpose of PR?**:

Fixes #1688 

**Does this PR introduce a breaking change?**

**If the changes in this PR are manually verified, list down the scenarios covered:**:

**Additional information for reviewer?** :
_Mention if this PR is part of any design or a continuation of previous PRs_

**Checklist:**
- [X] Bug fix. Fixes #<issue number>
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] PR Title follows the convention of  `<type>(<scope>): <subject>`
- [ ] Commit has unit tests
- [ ] Commit has integration tests

### CODE REVIEW

**Comment by yp969803** [comment]
@daemon1024 can you review the pr

**Comment by yp969803** [comment]
@daemon1024 getting 
```
Could not read path from procfs due to readlink /proc/0/exe: no such file or directory
```

but i have mounted the /proc directory in the kubearmor pod

**Comment by yp969803** [comment]
The error occurs here
```
	if data, err := os.Readlink("/proc/" + strconv.FormatUint(uint64(ctx.HostPPID), 10) + "/exe"); err == nil && data != "" && data != "/"
	```
i think we have to mount the "/proc" at an arribitary location scuh as /host/proc to avoid PID namespace isolation

**Comment by yp969803** [comment]
Getting the same thing while using HostPID
```
Could not read path from procfs due to readlink /proc/0/exe: no such file or directory
```

File: deployments/get/objects.go [hunk 1]
```diff
@@ -283,6 +283,11 @@ func GenerateDaemonSet(env, namespace string) *appsv1.DaemonSet {
 			MountPath: "/media/root/etc/os-release",
 			ReadOnly:  true,
 		},
+		{
+			Name:      "procfs-path", //BPF (read-only)
+			MountPath: "/proc",
+			ReadOnly:  true,
+		},
```
Comment (daemon1024) [inline deployments/get/objects.go:290]: We will need to mount it in a custom directory, maybe /host/proc and configure KubeArmor to consider the custom procfs path wherever it's using it.

Comment (yp969803) [reply deployments/get/objects.go:290]: @daemon1024 so i have to check if the env is k8s and if true i have to change "/proc" to "/host/proc" whereever the proc directory is used in the Kuberarmor

**Comment by daemon1024** [comment]
Closing in favor of https://github.com/kubearmor/KubeArmor/pull/1831

**Pull request closed without merging**